api/v1alpha2/conversion.go

```diff
@@ -79,13 +80,28 @@ func (dst *ClusterList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *Machine) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha3.Machine)
 
+	// Manually restore data.
+	restored := &v1alpha3.Machine{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); ok {
```

Comment:
Personally, I prefer this style:

```go
ok, err := utilconversion.UnmarshalData(src, restored)
if err != nil {
  return err
}
if ok {
  if restored.Spec.Bootstrap.DataSecretName != nil {
    dst.Spec.Bootstrap.DataSecretName = restored.Spec.Bootstrap.DataSecretName
  }
}

return Convert_v1alpha2_Machine_To_v1alpha3_Machine(src, dst, nil)
```

But you don't have to change it if you don't want to. However, I would recommend checking `err` first, then `ok` second.